hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -285,8 +324,8 @@ protected Option<T> merge(Option<T> older, Map<String, Object> olderInfoMap,
    *  1. A set of pre-specified keys exists.
    *  2. The key of the record is not contained in the set.
    */
-  protected boolean shouldSkip(T record, String keyFieldName, boolean isFullKey, Set<String> keys) {
-    String recordKey = readerContext.getValue(record, readerSchema, keyFieldName).toString();
+  protected boolean shouldSkip(T record, String keyFieldName, boolean isFullKey, Set<String> keys, Schema dataBlockSchema) {
```

Comment:
Is `dataBlockSchema` the writer schema?  Rename it as `writerSchema`?

Reply:
yep